Cleanup Remote Docker Server build step does not appear to be working correctly

I'm seeing Build Images steps for ARM builds that are picking up caches layers.  The only way this can happen is if the cleanup logic is not correctly cleaning up the arm build agents.

Example:
https://dev.azure.com/dnceng/internal/_build/results?buildId=663036&view=logs&j=2d7881e9-6614-5601-c4d2-9bd4d2552c8d&t=066a7672-8d09-5bfd-e93d-01716f04f615&l=157

```
-- EXECUTING: docker build -t dotnetdocker.azurecr.io/build-staging/663036/dotnet/core-nightly/sdk:2.1.806-stretch-arm32v7 -t dotnetdocker.azurecr.io/build-staging/663036/dotnet/core-nightly/sdk:2.1-stretch-arm32v7 -t dotnetdocker.azurecr.io/build-staging/663036/dotnet/core-nightly/sdk:2.1.806 -t dotnetdocker.azurecr.io/build-staging/663036/dotnet/core-nightly/sdk:2.1 -f 2.1/sdk/stretch/arm32v7/Dockerfile 2.1/sdk/stretch/arm32v7
Sending build context to Docker daemon  3.584kB

Step 1/6 : FROM arm32v7/buildpack-deps:stretch-scm
 ---> 2793efb40472
Step 2/6 : RUN apt-get update     && apt-get install -y --no-install-recommends         libc6         libgcc1         libgssapi-krb5-2         libicu57         liblttng-ust0         libssl1.0.2         libstdc++6         zlib1g     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 99c002f00e60
Step 3/6 : ENV DOTNET_SDK_VERSION 2.1.806
 ---> Using cache
 ---> 1b7a7e9c0838
Step 4/6 : RUN curl -SL --output dotnet.tar.gz https://dotnetcli.azureedge.net/dotnet/Sdk/$DOTNET_SDK_VERSION/dotnet-sdk-$DOTNET_SDK_VERSION-linux-arm.tar.gz     && dotnet_sha512='80b38700c57cd0af6b769db9c8b171914e99512909e1181a87f014ac45201b38080531af334c7a7b429497be754e5b648600a9d86763fc2874a2d99cea6568ad'     && echo "$dotnet_sha512 dotnet.tar.gz" | sha512sum -c -     && mkdir -p /usr/share/dotnet     && tar -zxf dotnet.tar.gz -C /usr/share/dotnet     && rm dotnet.tar.gz     && ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet     && curl -SL --output /usr/share/dotnet/sdk/$DOTNET_SDK_VERSION/nuGetPackagesArchive.lzma https://dotnetcli.azureedge.net/dotnet/Sdk/$DOTNET_SDK_VERSION/nuGetPackagesArchive.lzma     && lzma_sha512='dbed58ffa155e98b7ec177be4e53d8f1b03b31b40787a6626309d85a230a1fbf2e03d34fa64664c144a42c313a70a8579c7f5a0c2a0e6041691ebf5a3f5f74b3'     && echo "$lzma_sha512 /usr/share/dotnet/sdk/$DOTNET_SDK_VERSION/nuGetPackagesArchive.lzma" | sha512sum -c -
 ---> Using cache
 ---> 54eb9a6a6cdb
Step 5/6 : ENV ASPNETCORE_URLS=http://+:80     DOTNET_RUNNING_IN_CONTAINER=true     DOTNET_USE_POLLING_FILE_WATCHER=true     NUGET_XMLDOC_MODE=skip
 ---> Using cache
 ---> ab324229fd22
Step 6/6 : RUN dotnet help
 ---> Using cache
 ---> 89a1b3ce7b5e
```

The source of this appears to be an error in the `Cleanup Remote Docker Server (images)` step.

```
##[error]Bash exited with code '127'.
```

This is a regression caused by a recent change to use Alpine based ImageBuilder images - https://github.com/dotnet/docker-tools/pull/505

--- a/eng/common/templates/steps/cleanup-docker-linux.yml
+++ b/eng/common/templates/steps/cleanup-docker-linux.yml
@@ -16,7 +16,7 @@ steps:
     condition: always()
     continueOnError: true
   - script: >
-      docker run --rm $(dockerArmRunOptions) --entrypoint /bin/bash $(dockerClientImage) -c
+      docker run --rm $(dockerArmRunOptions) --entrypoint /bin/sh $(dockerClientImage) -c
       'docker rmi -f $(docker images --format "{{.Repository}}:{{.Tag}}" | grep -v -e ^arm32v7/ -e ^arm64v8/ -e ^mcr\\.microsoft\\.com/windows -e :\<none\>$)'
     displayName: Cleanup Remote Docker Server (images)
     condition: always()

